src/bridge/views/InviteWallet.js

```diff
@@ -0,0 +1,289 @@
+import KeysAndMetadata from './Point/KeysAndMetadata'
+import Actions from './Point/Actions'
+import { BRIDGE_ERROR } from '../lib/error'
+import { Row, Col, Warning, Button, H1, H3 } from '../components/Base'
+
+// for wallet generation
+import * as kg from '../../../node_modules/urbit-key-generation/dist/index'
+import * as more from 'more-entropy'
+import lodash from 'lodash'
+import PaperCollateralRenderer from 'PaperCollateralRenderer'
+import JSZip from 'jszip'
+import saveAs from 'file-saver'
+import { MAX_GALAXY, MIN_STAR, MAX_STAR, MIN_PLANET,
+         GALAXY_ENTROPY_BITS, STAR_ENTROPY_BITS, PLANET_ENTROPY_BITS,
+         SEED_ENTROPY_BITS,
+         GEN_STATES
+       } from '../../walletgen/lib/constants'
+const SEED_LENGTH_BYTES = SEED_ENTROPY_BITS / 8
+const NEXT_STEP_NUM = 6;
+
+class InviteWallet extends React.Component {
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      loaded: false,
+      point: Nothing(),
+      wallet: Nothing(),
+      paper: Nothing(),
+      downloaded: false
+    }
+
+    this.download = this.download.bind(this)
+    this.proceed = this.proceed.bind(this)
+  }
+
+  componentDidMount() {
+    const { web3, wallet, contracts, setPointCursor } = this.props
+
+    web3.chain(_ =>
+    contracts.chain(ctrcs =>
+    wallet.chain(async wal => {
+      const addr = addressFromSecp256k1Public(wal.publicKey);
+      console.log('fetching incoming for', addr)
+      const incoming = await azimuth.azimuth.getTransferringFor(ctrcs, addr)
+      console.log('incoming', incoming)
+
+      let point = Nothing();
+      if (incoming.length > 0) {
+        point = Just(incoming[0]);
+        this.generateWallet(incoming[0]);
+        this.setState({ loaded: true, point: point });
+        if (incoming.length > 1) {
+          //TODO  notify user "...and others / ticket reusable"
+        }
+      }
+    })))
+  }
+
+  componentDidUpdate(prevProps) {
+    //
+  }
+
+  //TODO pulled from walletgen/views/Generate and Download, put into lib
+  async generateWallet(point) {
+    const makeTicket = point => {
+
+      const bits = point < MIN_STAR
+        ? GALAXY_ENTROPY_BITS
+        : point < MIN_PLANET
+          ? STAR_ENTROPY_BITS
+          : PLANET_ENTROPY_BITS
+
+      const bytes = bits / 8
+      const some = new Uint8Array(bytes)
+      window.crypto.getRandomValues(some)
+
+      const gen = new more.Generator()
+
+      return new Promise((resolve, reject) => {
+        gen.generate(bits, result => {
+          const chunked = lodash.chunk(result, 2)
+          const desired = chunked.slice(0, bytes) // only take required entropy
+          const more = lodash.flatMap(desired, arr => arr[0] ^ arr[1])
+          const entropy = lodash.zipWith(some, more, (x, y) => x ^ y)
+          const buf = Buffer.from(entropy)
+          const patq = ob.hex2patq(buf.toString('hex'))
+          resolve(patq)
+          reject('Entropy generation failed')
```

Comment:
Does this `reject` ever get run? Maybe there's a condition where you want to reject it.

Reply:
I pulled this from walletgen's `Generate` view. Not sure why it was written like this, and I don't think that as-is it can ever `reject`? idk where this would ever fail, anyway.

I'll probably clean that out once I move this into a lib.